src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.props

```diff
@@ -1,22 +1,20 @@
 <Project>
 
-  <PropertyGroup Condition="'$(BlazorBuildReferenceFromSource)'==''">
-    <BlazorJsPath>$(MSBuildThisFileDirectory)../tools/blazor/blazor.*.js</BlazorJsPath>
+  <PropertyGroup>
+    <BlazorJsPath Condition="'$(BlazorJsPath)' == ''">$(MSBuildThisFileDirectory)..\tools\blazor\blazor.webassembly.js</BlazorJsPath>
   </PropertyGroup>
 
   <PropertyGroup Label="Blazor build outputs">
     <MonoLinkerI18NAssemblies>none</MonoLinkerI18NAssemblies> <!-- See Mono linker docs - allows comma-separated values from: none,all,cjk,mideast,other,rare,west -->
     <AdditionalMonoLinkerOptions>--disable-opt unreachablebodies --verbose --strip-security true --exclude-feature com --exclude-feature sre -v false -c link -u link -b true</AdditionalMonoLinkerOptions>
-    <BaseBlazorDistPath>dist/</BaseBlazorDistPath>
-    <BaseBlazorPackageContentOutputPath>$(BaseBlazorDistPath)_content/</BaseBlazorPackageContentOutputPath>
-    <BaseBlazorRuntimeOutputPath>$(BaseBlazorDistPath)_framework/</BaseBlazorRuntimeOutputPath>
-    <BaseBlazorRuntimeBinOutputPath>$(BaseBlazorRuntimeOutputPath)_bin/</BaseBlazorRuntimeBinOutputPath>
-    <BaseBlazorRuntimeWasmOutputPath>$(BaseBlazorRuntimeOutputPath)wasm/</BaseBlazorRuntimeWasmOutputPath>
-    <BaseBlazorJsOutputPath>$(BaseBlazorRuntimeOutputPath)</BaseBlazorJsOutputPath>
-    <BaseBlazorIntermediateOutputPath>blazor/</BaseBlazorIntermediateOutputPath>
-    <BlazorWebRootName>wwwroot/</BlazorWebRootName>
+    <BaseBlazorDistPath>dist\</BaseBlazorDistPath>
+    <BaseBlazorPackageContentOutputPath>$(BaseBlazorDistPath)_content\</BaseBlazorPackageContentOutputPath>
+    <BaseBlazorRuntimeOutputPath>$(BaseBlazorDistPath)_framework\</BaseBlazorRuntimeOutputPath>
+    <BlazorRuntimeBinOutputPath>$(BaseBlazorRuntimeOutputPath)_bin\</BlazorRuntimeBinOutputPath>
+    <BlazorRuntimeWasmOutputPath>$(BaseBlazorRuntimeOutputPath)wasm\</BlazorRuntimeWasmOutputPath>
+    <BlazorWebRootName>wwwroot\</BlazorWebRootName>
     <BlazorBootJsonName>blazor.boot.json</BlazorBootJsonName>
-    <BlazorBootJsonOutputPath>$(BaseBlazorRuntimeOutputPath)$(BlazorBootJsonName)</BlazorBootJsonOutputPath>
+    <_BlazorBuiltInBclLinkerDescriptor>$(MSBuildThisFileDirectory)BuiltInBclLinkerDescriptor.xml</_BlazorBuiltInBclLinkerDescriptor>
```

Comment:
Feel free to say you don't want to deal with this question, since it's a bit tangential, but I'll ask now since you're thinking about stuff in this area.

We've had a request from Mono to make it easier to build Blazor apps against a Mono build in a custom directory, instead of the Mono build bundled into our `Microsoft.AspNetCore.Blazor.Mono.nupkg`.

Given an artifact like https://jenkins.mono-project.com/job/test-mono-mainline-wasm/4237/label=ubuntu-1804-amd64/Azure/processDownloadRequest/4237/ubuntu-1804-amd64/sdks/wasm/mono-wasm-b8fe43da3ed.zip extracted into a folder, could a third-party add some properties to their Blazor application (not in our repo) to set paths that would cause the build to pull in `mono.js/wasm` and the Mono BCL from that external location?

They want this because it will make it far easier for them to test the effects of Mono changes on Blazor, without having to build Blazor itself from sources.

Reply:
The ultimate ideal would be adding something like:

```
<DotNetWebAssemblySDKRoot>c:\me\blah\mono-wasm-b8fe43da3ed\</DotNetWebAssemblySDKRoot>
```

... and having that alone be sufficient to use this custom build of Mono WebAssembly.

Do you think that's a separate future feature, or is it easy to inline with the work you're doing here?

Reply:
Named it `DotNetWebAssemblyArtifactsRoot`. But yes that's pretty easy to do.